tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -17,32 +17,46 @@ var _ = Describe("odo devfile push command tests", func() {
 	var namespace, context, cmpName, currentWorkingDirectory, projectDirPath string
 	var projectDir = "/projectDir"
 	var sourcePath = "/projects/nodejs-web-app"
+	var clusterType helper.CliRunner
 
-	// TODO: all oc commands in all devfile related test should get replaced by kubectl
-	// TODO: to goal is not to use "oc"
-	oc := helper.NewOcRunner("oc")
+	// Using program commmand according to cluter type in devfile
+	if os.Getenv("KUBERNETES") == "true" {
+		clusterType = helper.NewKubectlRunner("kubectl")
+	} else {
+		clusterType = helper.NewOcRunner("oc")
+	}
 
 	// This is run after every Spec (It)
 	var _ = BeforeEach(func() {
 		SetDefaultEventuallyTimeout(10 * time.Minute)
-		namespace = helper.CreateRandProject()
 		context = helper.CreateNewContext()
-		currentWorkingDirectory = helper.Getwd()
 		projectDirPath = context + projectDir
-		cmpName = helper.RandString(6)
-
-		helper.Chdir(context)
-
 		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
 
 		// Devfile push requires experimental mode to be set
 		helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+
+		if os.Getenv("KUBERNETES") == "true" {
+			homeDir := helper.GetUserHomeDir()
+			kubeConfigFile := helper.CopyKubeConfigFile(filepath.Join(homeDir, ".kube", "config"), filepath.Join(context, "config"))
+			namespace = helper.CreateRandNamespace(kubeConfigFile)
+		} else {
+			namespace = helper.CreateRandProject()
+		}
+		currentWorkingDirectory = helper.Getwd()
+		cmpName = helper.RandString(6)
+		helper.Chdir(context)
 	})
 
 	// Clean up after the test
 	// This is run after every Spec (It)
 	var _ = AfterEach(func() {
-		helper.DeleteProject(namespace)
+		if os.Getenv("KUBERNETES") == "true" {
```

Comment:
This condition should also be extracted into a function.

Reply:
https://github.com/openshift/odo/pull/3041#discussion_r421446577

Reply:
Why don't we create something like  `helper.DeleteNamespaceOrProject` (not a good name, this is just for illustration) to avoid this whole `if` statement?

in `OcRunner` this function would be implemented in a way that it deletes Project.
in `KubectlRunner` this function would be implemented in a way that it deletes Namespace

The same could be don for `CreateRandProject`/`CreateRandNamespace`

Reply:
> Why don't we create something like `helper.DeleteNamespaceOrProject` (not a good name, this is just for illustration) and make it part of `CliRunner` interface to avoid this whole `if` statement?
> 
> in `OcRunner` this function would be implemented in a way that it deletes Project.
> in `KubectlRunner` this function would be implemented in a way that it deletes Namespace
> 
> The same could be done for `CreateRandProject`/`CreateRandNamespace`

I will check and figure it out.

Reply:
https://github.com/openshift/odo/issues/2752#issuecomment-627977945

Reply:
> Why don't we create something like helper.DeleteNamespaceOrProject (not a good name, this is just for illustration) and make it part of CliRunner interface to avoid this whole if statement?

@kadel Yes creating `helper.DeleteNamespaceProject` and `helper.CreateRanNamespaceProject` make sense. However I would say the tests which are only running for openshift let it use `helper.CreateRanProject` and `helper.DeleteProject` because replacing it with `helper.CreateRanNamespaceProject` won't justify the running cluster whereas we can implement `helper.DeleteNamespaceProject` and `helper.CreateRanNamespaceProject` for devfile related tests only using `CliRunner`.

IMO changing the project create/delete function name will only justify devfile related test. So lets keep it as it is for the tests only running on openshift and implement the changes for the devfile tests.

Reply:
Addressed the requested change in 73c4a24